azurerm/resource_arm_autoscale_setting.go

```diff
@@ -432,7 +432,9 @@ func resourceArmAutoScaleSettingRead(d *schema.ResourceData, meta interface{}) e
 		return fmt.Errorf("Error setting `notification` of Autoscale Setting %q (resource group %q): %+v", name, resourceGroup, err)
 	}
 
-	flattenAndSetTags(d, resp.Tags)
+	// Return a new tag map filtered by the specified tag names.
+	tagMap := filterTags(resp.Tags, "$type")
```

Comment:
Should this filtering be applied more globally? There seems to have been a few issues related to it across various resources.

Reply:
Not necessarily unfortunately - this field's only applicable to certain API's (read: Monitor/App Insights) - such that it's potentially a valid key in other API's